### ISSUE TITLE
Explicitly state that UA can reject file based on its type

### ISSUE THREAD

**Issue by mgiuca** [w3c/web-share#127]
With examples to prevent sharing of executable files.

Could be based on extension, MIME type or contents.

**Comment by mgiuca** [comment]
It actually does have this (search for the word "block").

But it needs to change to split it out from the user activation check, and move it into the "in parallel" steps, so it can be done async.

**Comment by marcoscaceres** [comment]
Over on the gecko side, we've been considering the following attack:

```JS
const evil = new File([...evilBytes], "cat.exe");
navigator.share({files: [evil], title: "Cute cat image!"});
```

In order to mitigate the above, we should require each `File` have both type and a filename with an extension, and that the type must match an expected extension. We should have a limited list of recommended sharable types/extension combinations. 

This also allows us to avoid having to do content type sniffing, although that could be done as well. 

So, concretely, we should replace:

> If a file type is being blocked due to security considerations, return a promise rejected with with a "NotAllowedError" DOMException.

With a lookup-table for allowed MIME + extensions-list combinations.

**Comment by mgiuca** [comment]
TPAC 2020: Resolved to make a change to the spec to let browsers determine their own rules for blocking based on file type.

We can discuss further having a formal list of allowed / blocked types, but no consensus on that.

@NotWoods suggests https://html.spec.whatwg.org/multipage/links.html#allowed-to-download as a starting point.

**Comment by marcoscaceres** [comment]
The HTML suggestion relates to the `download=` attribute, but in our case we use Permissions Policy to enable the functionality. However, I adapted the text "Optionally, the user agent may return false, if it believes doing so would safeguard the user from a potentially hostile download." and added it to https://github.com/w3c/web-share/pull/185/commits/6c0a9f5a90a4a35375f3211c385be380c73b5a62.

**Comment by marcoscaceres** [comment]
List of extensions / types from Chromium : https://docs.google.com/document/d/1tKPkHA5nnJtmh2TgqWmGSREUzXgMUFDL6yMdVZHqUsg/edit

**Comment by marcoscaceres** [comment]
closing as handled by https://github.com/w3c/web-share/commit/6c0a9f5a90a4a35375f3211c385be380c73b5a62 for now.
